lib/bundler/rubygems_gem_installer.rb

```diff
@@ -12,5 +12,33 @@ def self.at(*args)
     def check_executable_overwrite(filename)
       # Bundler needs to install gems regardless of binstub overwriting
     end
+
+    def pre_install_checks
+      super && validate_bundler_checksum(options[:bundler_expected_checksum])
+    end
+
+  private
+
+    def validate_bundler_checksum(checksum)
+      return true if Bundler.settings[:disable_checksum_validation]
```

Comment:
none of the other bool keys have dots in them, and wouldn't that break doing the setting as an env var?